### DISCUSSION TITLE
About 5GMS and MBMS

### DISCUSSION THREAD

**Discussion by mSerendipity** [5G-MAG/Getting-Started#74] [Q&A] [answered]
Hello, I saw a figure of 5GMS combined with MBMS, similar to the 3GPP TS 26.501 4.6 5G Downlink Media Streaming via eMBMS. May I ask if this feature is implemented in 5G-MAG and if the 5GMS-Aware Application can receive broadcast content from Middleware?
![5G-MAG RT 5G Architecture v6](https://github.com/5G-MAG/Getting-Started/assets/82158989/5158c9b5-f3ca-4370-8ea6-e807f280cd48)

**Comment by dsilhavy** [comment, marked as answer]
Right now, our 5GMS-Aware Application is not integrated into the 5GBC stack. Ultimately, it is the goal to combine both the unicast and the broadcast world, as illustrated in your figure above.

What we have in place for now are dedicated components to handle broadcast reception. The implementation also supports a switch between broadcast and unicast, but this is not integrated into the 5GMS architecture. Instead, the unicast part is "plain" OTT streaming. Here is the relevant link to the documentation: https://5g-mag.github.io/Getting-Started/pages/lte-based-5g-broadcast/

Here are the Github projects:

## Linux stack
* https://github.com/5G-MAG/rt-mbms-modem
* https://github.com/5G-MAG/rt-mbms-mw
* https://github.com/5G-MAG/rt-wui
* https://github.com/5G-MAG/rt-mbms-tx

## Android stack
* https://github.com/5G-MAG/rt-mbms-tx-for-qrd-and-crd
* https://github.com/5G-MAG/rt-mbms-mw-android (not yet released but early access can be requested here: https://www.5g-mag.com/early-access)

**Comment by mSerendipity** [comment]
Thank you very much for your reply! I've tried building these projects before, when running a sample file, the middleware worked well:
![middleware1](https://github.com/5G-MAG/Getting-Started/assets/82158989/4a7eb85c-4102-452b-9901-236a46696dfb)
But when trying to receive broadcast signals sent by rt-mbms-tx, middleware either reported an error or crashed.
![middleware2](https://github.com/5G-MAG/Getting-Started/assets/82158989/3d9a253c-cd2d-4c9f-95c8-37ca19b727d7)
![middleware3](https://github.com/5G-MAG/Getting-Started/assets/82158989/42a88de0-adc2-45b3-8769-159b69ad9c57)
I guess it may be a configuration issue with the SA file in Middleware. May I ask how to achieve the effect shown in the demonstration video of Seamless Switching?
